staging/src/k8s.io/client-go/util/certificate/certificate_manager.go

```diff
@@ -158,9 +170,17 @@ func NewManager(config *Config) (Manager, error) {
 		return nil, err
 	}
 
+	getTemplate := config.GetTemplate
+	dynamicTemplate := true
+	if getTemplate == nil {
+		getTemplate = func() *x509.CertificateRequest { return config.Template }
+		dynamicTemplate = false
+	}
+
 	m := manager{
 		certSigningRequestClient: config.CertificateSigningRequestClient,
-		template:                 config.Template,
+		getTemplate:              getTemplate,
+		dynamicTemplate:          dynamicTemplate,
```

Comment:
This can be `config.GetTemplate != nil` and local `dynamicTemplate` var can be removed

Reply:
config.GetTemplate and config.Template are mutually exclusive. `dynamicTemplate` is used to determine whether we need to monitor getTemplate() for changes